### PR TITLE
odbc: conan v2 support + unvendor libltdl

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
```
CMakeFiles/test_package.dir/build.make:89: recipe for target 'test_package' failed
CMakeFiles/Makefile2:75: recipe for target 'CMakeFiles/test_package.dir/all' failed
Makefile:83: recipe for target 'all' failed
ld: error: duplicate symbol: __get_connection
>>> defined at SQLGetDiagRec.c
>>>            SQLGetDiagRec.o:(__get_connection) in archive /home/conan/w/prod/BuildSingleReference/.conan/data/odbc/2.3.7/_/_/package/8eabf8030762d318d462d63ad71a06248b66af41/lib/libodbccr.a
>>> defined at SQLGetDiagRec.c
>>>            SQLGetDiagRec.o:(.text+0x10) in archive /home/conan/w/prod/BuildSingleReference/.conan/data/odbc/2.3.7/_/_/package/8eabf8030762d318d462d63ad71a06248b66af41/lib/libodbc.a
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

Weird, there are ODR violations in test_package but not in test_v1_package...

**Review by uilianries — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/odbc/all/conandata.yml [hunk 1]
```diff
@@ -15,6 +15,10 @@ sources:
...
 patches:
+  "2.3.11":
+    - patch_file: "patches/0001-duplicated-get-connection.patch"
+  "2.3.9":
+    - patch_file: "patches/0001-duplicated-get-connection.patch"
   "2.3.7":
-    - patch_file: "patches/missing-declarations.patch"
-      base_path: "source_subfolder"
+    - patch_file: "patches/0001-duplicated-get-connection.patch"
+    - patch_file: "patches/0002-missing-declarations.patch"
```
Comment (prince-chrismc) [inline recipes/odbc/all/conandata.yml:24]: Patch fields 🙏

File: recipes/odbc/all/conanfile.py [hunk 2]
```diff
@@ -27,84 +29,106 @@ class OdbcConan(ConanFile):
...
+            libiconv_prefix = self.dependencies["libiconv"].package_folder
+            tc.configure_args.append(f"--with-libiconv-prefix={libiconv_prefix}")
+        tc.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+        # support more triplets
         shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
```
Comment (prince-chrismc) [inline recipes/odbc/all/conanfile.py:92]: Files copy?

**Review by prince-chrismc — APPROVED** [review]

File: recipes/odbc/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/odbc/all/conanfile.py:92]: How?

Comment (uilianries) [reply recipes/odbc/all/conanfile.py:92]: The new `tools.copy` it does not so dummy like `self.copy`:
https://docs.conan.io/en/latest/reference/conanfile/tools/files/basic.html#conan-tools-files-copy

Comment (prince-chrismc) [reply recipes/odbc/all/conanfile.py:92]: Ideal the Conan api should be doing file io `tools.files` api

**Pull request merged**